File: frontend/src/types/transaction.ts
// types/transaction.ts
// All project types in one place

export type TransactionStatus = 'Pending' | 'Completed' | 'Failed';

export interface Transaction {
  transactionId: string;
  amount: number;
  currency: string;
  status: TransactionStatus;
  timestamp: string; // ISO 8601 string
}

// Status enum for filters
export const TRANSACTION_STATUSES: TransactionStatus[] = ['Pending', 'Completed', 'Failed'];

// Colors by status - centralized theme
export const STATUS_CONFIG: Record<TransactionStatus, { 
  color: string; 
  bg: string; 
  label: string 
}> = {
  Pending:   { color: '#f59e0b', bg: 'rgba(245,158,11,0.12)',  label: 'PENDING'   },
  Completed: { color: '#10b981', bg: 'rgba(16,185,129,0.12)',  label: 'COMPLETED' },
  Failed:    { color: '#ef4444', bg: 'rgba(239,68,68,0.12)',   label: 'FAILED'    },
};

File: backend/FinancialMonitor.API/Models/Transaction.cs
using System.Text.Json.Serialization;

namespace FinancialMonitor.API.Models;

/// <summary>
/// Transaction status enum — three possible states.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum TransactionStatus
{
    Pending,
    Completed,
    Failed
}

/// <summary>
/// The central domain entity.
/// NOTE: We keep this as a class (not record) because EF Core
/// tracks entities by reference and uses change-tracking internally.
/// Records with value semantics can cause subtle EF bugs.
/// </summary>
public class Transaction
{
    /// <summary>Unique identifier — must be a valid GUID</summary>
    public required string TransactionId { get; set; }

    /// <summary>Transaction amount — must be positive</summary>
    public decimal Amount { get; set; }

    /// <summary>Currency code (USD, EUR, ILS, etc.)</summary>
    public required string Currency { get; set; }

    /// <summary>Current status of the transaction</summary>
    public TransactionStatus Status { get; set; }

    /// <summary>Transaction time — ISO 8601</summary>
    public DateTime Timestamp { get; set; }
}
